tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/passes/BoundedCheckerPassImpl.scala

```diff
@@ -60,28 +60,160 @@ class BoundedCheckerPassImpl @Inject() (val options: PassOptions,
-    val tuning = options.getOrElse("general", "tuning", Map[String, String]())
-    val checkRuntime = false // deprecated, will be removed in 0.7.0
-    val strategy =
-      if (search == "bfs") {
-        new BfsStrategyStopWatchDecorator(new BfsStrategy(input, stepsBound), filename="bfs.csv")
-      } else {
-        val random = tuning.getOrElse("search.randomDfs", "")
-        new DfsStrategy(input, stepsBound, random.toLowerCase.equals("true"))
-      }
+    val solverContext: RecordingSolverContext = RecordingSolverContext.createZ3(None, solverConfig)
 
-    val checker: Checker =
-        new ModelChecker(typeFinder, hintsStore, changeListener, exprGradeStore, sourceStore,
-          input, strategy, tuning, debug, profile, checkRuntime)
+    val typeFinder = new TrivialTypeFinder
+    val rewriter: SymbStateRewriterImpl = new SymbStateRewriterImpl(solverContext, typeFinder, exprGradeStore)
+    rewriter.formulaHintsStore = hintsStore
+    rewriter.config = RewriterConfig(tuning)
 
+    type SnapshotT = IncrementalExecutionContextSnapshot
+    val executorContext = new IncrementalExecutionContext(rewriter)
+    val trex =
+      new TransitionExecutorImpl[SnapshotT](params.consts, params.vars, executorContext)
+    val filteredTrex =
+      new FilteredTransitionExecutor[SnapshotT](params.transitionFilter, params.invFilter, trex)
+
+    val checker = new SeqModelChecker[SnapshotT](params, input, filteredTrex)
     val outcome = checker.run()
-    logger.info("The outcome is: " + outcome)
-    outcome == Checker.Outcome.NoError
+    logger.info(s"The outcome is: " + outcome)
+    outcome == Outcome.NoError
+  }
+
+  private def runOfflineChecker(params: ModelCheckerParams,
+                                input: CheckerInput,
+                                tuning: Map[String, String],
+                                solverConfig: SolverConfig
+                               ): Boolean = {
+    val profile = options.getOrElse("smt", "prof", false)
+    val solverContext: RecordingSolverContext = RecordingSolverContext.createZ3(None, solverConfig)
+
+    val typeFinder = new TrivialTypeFinder
+    val rewriter: SymbStateRewriterImpl = new SymbStateRewriterImpl(solverContext, typeFinder, exprGradeStore)
+    rewriter.formulaHintsStore = hintsStore
+    rewriter.config = RewriterConfig(tuning)
+
+    type SnapshotT = OfflineExecutionContextSnapshot
+    val executorContext = new OfflineExecutionContext(rewriter)
+    val trex = new TransitionExecutorImpl[SnapshotT](params.consts, params.vars, executorContext)
+    val filteredTrex = new FilteredTransitionExecutor[SnapshotT](params.transitionFilter, params.invFilter, trex)
+
+    val checker = new SeqModelChecker[SnapshotT](params, input, filteredTrex)
+    val outcome = checker.run()
+    logger.info(s"The outcome is: " + outcome)
+    outcome == Outcome.NoError
+  }
+
+  /*
+  TODO: the following code will be uncommented when the paralel checker is transferred from ik/multicore
```

Comment:
Could this behind a feature flag or left for a separate PR instead? Feels wrong to me to commit commented out code on principle. Of course, I won't block on it if there's no other way.

Reply:
It will not compile at the moment. But the next PR will make it compilable :-)

Reply:
So shouldn't it be part of the next PR then? ;)

Reply:
in the ideal world, yes, there should be no commented code. But as I am copying over files, it is better this way :-)